### ISSUE TITLE
Pipeline metadata json output should sanitize/encode VLR description field.

### ISSUE THREAD

**Issue by bacek** [PDAL/PDAL#3829]
Using `pdal pipeline --metadata meta.json` produces broken JSON when LAS file has VLRs with non-string "description".

```
$ pdal -v 8 translate --metadata meta.json in.las out.laz
(PDAL Debug) Debugging...
(pdal translate Debug) Executing pipeline in stream mode.
(pdal pipeline writers.las Debug) Wrote 138308654 points to the LAS file

$ jq . < meta.json 
parse error: Invalid escape at line 81, column 54
```

Relevant part of meta.json

```
      "vlr_1":
      {
        "data": "MDAwMDAwMDAwMDAwMDAwMQ==",
        "description": "0000000000000001�\u0014\0012",
        "record_id": 4711,
        "user_id": "guid"
      },
```

Just an empty pipeline with  `in.las` VLRs. 

```
{
  "pipeline":[
    "in.las",
    "out.las"
  ]
}
```

**Expected behavior**
Pipeline metadata JSON should be JSON.

**System/installation information:**

```
$ uname -a
Linux sunburst 5.18.0-3-amd64 #1 SMP PREEMPT_DYNAMIC Debian 5.18.14-1 (2022-07-23) x86_64 GNU/Linux
```

```
$ pdal --version
pdal 2.2.0 (git-version: Release)
```

**Comment by hobu** [comment]
This was fixed at https://github.com/PDAL/PDAL/pull/3773 and is available in the 2.4.2 release

**Comment by bacek** [comment]
It's a different issue.

```
$ ~/src/PDAL/build/bin/pdal -v 8 translate MLN_ZSS.las out.laz --metadata m3.json
(PDAL Debug) Debugging...
(pdal translate Debug) Executing pipeline in stream mode.
(pdal translate writers.las Debug) Wrote 138308654 points to the LAS file

$ jq . < m3.json 
parse error: Invalid escape at line 64, column 54

$ ~/src/PDAL/build/bin/pdal --version
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
pdal 2.4.2 (git-version: 32235d)
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

```

xxd output:
```
000005c0: 2020 2020 7d2c 0a20 2020 2020 2022 766c      },.      "vl
000005d0: 725f 3122 3a0a 2020 2020 2020 7b0a 2020  r_1":.      {.  
000005e0: 2020 2020 2020 2264 6174 6122 3a20 224d        "data": "M
000005f0: 4441 774d 4441 774d 4441 774d 4441 774d  DAwMDAwMDAwMDAwM
00000600: 4441 774d 513d 3d22 2c0a 2020 2020 2020  DAwMQ==",.      
00000610: 2020 2264 6573 6372 6970 7469 6f6e 223a    "description":
00000620: 2022 3030 3030 3030 3030 3030 3030 3030   "00000000000000
00000630: 3031 e3a7 5c75 3030 3134 5c30 3031 3222  01..\u0014\0012"
00000640: 2c0a 2020 2020 2020 2020 2272 6563 6f72  ,.        "recor
00000650: 645f 6964 223a 2034 3731 312c 0a20 2020  d_id": 4711,.   
00000660: 2020 2020 2022 7573 6572 5f69 6422 3a20       "user_id": 
00000670: 2267 7569 6422 0a20 2020 2020 207d 2c0a  "guid".      },.

```

**Comment by hobu** [comment]
Can you post the file somewhere for me to test?

**Comment by bacek** [comment]
I generated one with
```
import laspy
import numpy as np


class wrapped:
    def encode(self, codec):
        return b'\x30\x30\x30\x30\x30\x30\x30\x30\x30\x30\x30\x30\x30\x30\x30\x31\xe3\xa7\x5c\x14\x12'

header = laspy.LasHeader(point_format=7, version="1.4")
header.add_extra_dim(laspy.ExtraBytesParams(name="random", type=np.int32))
header.offsets = np.array([0.0, 0.0, 0.0])
header.scales = np.array([0.1, 0.1, 0.1])

las = laspy.LasData(header)

new_vlr = laspy.VLR("The User ID",
                    4711,
                    description = wrapped(),
                    record_data=b"\x00" * 10,)
las.vlrs.append(new_vlr)
[test.zip](https://github.com/PDAL/PDAL/files/9214369/test.zip)

las.write("new_file.las")
```

**Comment by bacek** [comment]
```
$ python gen.py 
$ ls -l new_file.las 
-rw-r--r-- 1 bacek bacek 685 Jul 29 07:35 new_file.las
$ pdal translate new_file.las new_out.laz --metadata m5.json 
(pdal translate readers.las Error) Global encoding WKT flag not set for point format 6 - 10.
$ jq < m5.json 
parse error: Invalid escape at line 64, column 56

```

**Comment by abellgithub** [comment]
https://github.com/PDAL/PDAL/issues/3772

**Comment by hobu** [comment]
Which version of laspy are you using for this script? Using an updated version of your script, the most recent fails for me, where laspy is trying to force the description to ascii. This has been the LAS convention for nearly two decades, so what you're specifically doing here is likely to cause trouble for more than just PDAL.

```
import laspy
import numpy as np




desc_bytes = b'\x30\x30\x30\x30\x30\x30\x30\x30\x30\x30\x30\x30\x30\x30\x30\x31\xe3\xa7\x5c\x14\x12'

header = laspy.LasHeader(point_format=7, version="1.4")
header.add_extra_dim(laspy.ExtraBytesParams(name="random", type=np.int32))
header.offsets = np.array([0.0, 0.0, 0.0])
header.scales = np.array([0.1, 0.1, 0.1])

las = laspy.LasData(header)

new_vlr = laspy.VLR("The User ID",
                    4711,
                    description = desc_bytes,
                    record_data=b"\x00" * 10,)
las.vlrs.append(new_vlr)

las.write("unicode-fail.las")
```

```
  File "/Users/hobu/miniforge3/envs/pdal-build/lib/python3.9/site-packages/laspy/utils.py", line 66, in write_as_c_string
    _ = string.decode(encoding, errors=encoding_errors)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe3 in position 16: ordinal not in range(128)
```

**Comment by bacek** [comment]
This is `wrapper` class for. To generate incorrect bytes sequence. 

Original file I spotted issue with is quite large, and this code just produce minimal test case.

**Comment by bacek** [comment]
I actually attached test.zip to the original comment with generated broken LAS file.

**Comment by hobu** [comment]
A few comments:

* You shouldn't put non-ascii data in VLR descriptions. Newer laspy is trying prevent you from doing this, and other LAS-consuming software is likely to choke on it.
* PDAL's escapeJSON should be updated pitch characters so it writes valid JSON

**Comment by bacek** [comment]
I know that I shouldn't. And I don't put it deliberately. This was in a file received from our customer. If pdal will fail loading this file it will be acceptable solution.

**Comment by hobu** [comment]
Hopefully addressed in #3906. Please reopen if PDAL 2.5.0 doesn't fix this.
